pkgs/top-level/aliases.nix

```diff
@@ -2,9 +2,21 @@ self:
 
 with self;
 
+let
+  # Removing recurseForDerivation prevents derivations of aliased attribute
+  # set to appear while listing all the packages available.
+  removeRecurseForDerivations = _n: alias: with lib;
+    if alias.recurseForDerivations or false then
+      removeAttrs alias ["recurseForDerivations"]
+    else alias;
```

Comment:
https://github.com/NixOS/nix/blob/master/src/libexpr/primops.cc#L1060-L1080

Yes, the attribute will contain the same content, but this would be a new one.
I guess this would be easy to fix in Nix.